torch/_dynamo/variables/builder.py

```diff
@@ -1059,17 +1059,20 @@ def wrap_to_fake_tensor_and_record(
     ):
         static_shapes, reason = tensor_always_has_static_shape(e, source, is_tensor)
 
+        dynamic_indices = None
+        if hasattr(e, "_dynamo_dynamic_indices"):
```

Comment:
This part is still stateful, and read off the tensor. Perhaps there is a way to make it stateless here too?